### PR TITLE
AIRNEIS-43: Update Go files and add models

### PR DESCRIPTION
**Pull request by BarnabePILLIAUDIN**
# Done

Added the structs of the model
Replaced Price by Price with taxes and Price without taxes to help us when we will do invoices
Added a field isActive to address for history purpose so the previous orders are not affected when the address changes
Updated doc

# CLOSED WILL BE MERGED IN ANOTHER PR

### CODE REVIEW

File: back/model/Status.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package model
+
+import "gorm.io/gorm"
+
+type Status struct {
+	gorm.Model
+	Name    string `json:"name"`
+	IsOrder bool   `json:"isOrder" default:"false"`
+}
```
Comment (avetisk) [inline back/model/Status.go:9]: What is this model? What's its purpose?

File: back/model/Product.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+type Product struct {
+	gorm.Model
+	Name                    string         `json:"name"`
+	Slug                    string         `json:"slug" gorm:"unique"`
+	CategoryID              uint           `json:"categoryID"`
+	Category                Category       `json:"category"`
+	PriceWithoutTaxes       int            `json:"PriceWithoutTaxes"`
+	PriceWithTaxes          int            `json:"priceWithTaxes"`
+	Description             string         `json:"description"`
+	ThumbnailUrl            string         `json:"thumbnailUrl"`
```
Comment (avetisk) [inline back/model/Product.go:14]: Do you mean URI/ID or URL?

File: back/model/Product.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	OutOfStock              string         `json:"outOfStock" gorm:"default:false"`
```
Comment (avetisk) [inline back/model/Product.go:15]: Why a string?

File: back/model/Product.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import "gorm.io/gorm"
+
+type Product struct {
+	gorm.Model
+	Name                    string         `json:"name"`
+	Slug                    string         `json:"slug" gorm:"unique"`
+	CategoryID              uint           `json:"categoryID"`
+	Category                Category       `json:"category"`
+	PriceWithoutTaxes       int            `json:"PriceWithoutTaxes"`
+	PriceWithTaxes          int            `json:"priceWithTaxes"`
```
Comment (avetisk) [inline back/model/Product.go:12]: You should not store computed values in DB.

File: back/model/Status.go (hunk 1 above)
Comment (barnabepilliaudin) [reply back/model/Status.go:9]: It's to know if the the cart is the current cart of the use or a previous order. It allows to store everything in the same table

File: back/model/Product.go (hunk 2 above)
Comment (barnabepilliaudin) [reply back/model/Product.go:14]: It's the URL (we changed the typo later)

File: back/model/Status.go (hunk 1 above)
Comment (avetisk) [reply back/model/Status.go:9]: If I understand you right, then it's a bad design (no point lost though).

Comment (barnabepilliaudin) [reply back/model/Status.go:9]: Would have it been better to have a column isOrder in every cart?

File: back/model/Product.go (hunk 2 above)
Comment (avetisk) [reply back/model/Product.go:14]: 🆗

File: back/model/Status.go (hunk 1 above)
Comment (avetisk) [reply back/model/Status.go:9]: The issue is the name: `Status`.

Status of what?

Comment (barnabepilliaudin) [reply back/model/Status.go:9]: It's the status of the order it could be:
- Cart
- Preparation in progress
- Delivery in progress
- Delivered
- Cancelled

Comment (avetisk) [reply back/model/Status.go:9]: Then name it so! `OrderStatus`.

Moreover, if it's "just an enum", then why is it a full fledged entity?

File: back/model/Product.go (hunk 3 above)
Comment (barnabepilliaudin) [reply back/model/Product.go:15]: It's a fail it has been corrected later

**Pull request closed without merging**